robottelo/vm.py

```diff
@@ -232,19 +235,50 @@ def create(self):
             ping_from_hostname = settings.server.hostname
 
         # Give some time to machine boot
-        result = ssh.command(
-            u'for i in {{1..60}}; do ping -c1 {0}.local && exit 0; sleep 1;'
-            u' done; exit 1'.format(self._target_image),
+        qemu_ga_check = ssh.command(
+            u'virsh qemu-agent-command {0} '
+            '\'{{"execute":"guest-network-get-interfaces"}}\''.format(self.hostname),
             ping_from_hostname,
             connection_timeout=30
         )
-        if result.return_code != 0:
-            logger.error('Failed to obtain VM IP, reverting changes')
-            self.destroy()
-            raise VirtualMachineError(
-                'Failed to fetch virtual machine IP address information')
-        output = ''.join(result.stdout)
-        self.ip_addr = output.split('(')[1].split(')')[0]
+        if qemu_ga_check.return_code != 0:
+            if 'guest agent is not connected' in qemu_ga_check.stderr:
+                for i in range(60):
+                    qemu_ga_check = ssh.command(
+                        u'virsh qemu-agent-command {0} '
+                        '\'{{"execute":"guest-network-get-interfaces"}}\''.format(self.hostname),
+                        ping_from_hostname,
+                        connection_timeout=30
+                    )
+                    if qemu_ga_check.return_code != 0:
+                        sleep(1)
+                    else:
+                        ifaces = json.loads(qemu_ga_check.stdout[0])
+                        mgmt_if = [i for i in ifaces['return']
+                                   if 'eth0' in i['name'] or 'ens3' in i['name']]
+                        if len(mgmt_if) > 0:
+                            if mgmt_if[0].get('ip-addresses'):
+                                # get only the ipv4 addresses
+                                ip4_addr = [i['ip-address'] for i in mgmt_if[0]['ip-addresses']
+                                            if i['ip-address-type'] == 'ipv4']
+                                if len(ip4_addr) > 0:
+                                    self.ip_addr = ip4_addr[0]
+                                    break
+        else:
+            # fallback to avahi in case of any issue with qemu-guest-agent
```

Comment:
As mentioned in previous comment, we enter this path based only on response of initial `virsh qemu-agent-command` run. If anything happens in subsequent call, we do not execute this path.

I think this `else` was supposed to be on different indentation level than it actually is.